Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2626,6 +2630,36 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+bool CSSPropertyParser::consumeScrollTimelineShorthand(bool important)
+{
+    CSSValueListBuilder namesList;
+    CSSValueListBuilder axesList;
+
+    do {
+        // A valid scroll-timeline-name is required.
+        if (auto name = CSSPropertyParsing::consumeSingleScrollTimelineName(m_range))
+            namesList.append(name.releaseNonNull());
+        else
+            return false;
+
+        // A scroll-timeline-axis is optional.
+        if (m_range.peek().type() != IdentToken)
```

Comment:
```suggestion
        if (m_range.peek().type() == CommaToken)
```
I'd be more specific here.

Reply:
`if (m_range.peek().type() == CommaToken || m_range.atEnd())` is probably the condition you want.

Reply:
That does seem more comprehensive, I'll adopt it in the next iteration of this patch.